### PR TITLE
[bitnami/grafana-tempo] Refactor grafana-tempo cache configuration

### LINKED ISSUE #24164: [bitnami/grafana-tempo] Grafana Tempo chart sets deprecated configuration
### Name and Version

bitnami/grafana-tempo 2.11.0

### What architecture are you using?

amd64

### What steps will reproduce the bug?

Installing the chart out of the box leads to warnings printed to the logs.

In Tempo v2.4, cache configuration was refactored and the old configuration methods have been deprecated.

https://grafana.com/docs/tempo/latest/release-notes/v2-4/#cache-configuration-refactored

### Are you using any custom parameters or values?

None

### What is the expected behavior?

_No response_

### What do you see instead?

```
level=warn ts=2024-03-05T14:06:15.05609389Z caller=main.go:130 msg="-- CONFIGURATION WARNINGS --"
level=warn ts=2024-03-05T14:06:15.05616032Z caller=main.go:136 msg="c.StorageConfig.Trace.Cache is deprecated and will be removed in a future release." explain="Please migrate to the top level cache settings config."
level=info ts=2024-03-05T14:06:15.056186335Z caller=main.go:225 msg="initialising OpenTracing tracer"
level=info ts=2024-03-05T14:06:15.069045124Z caller=main.go:118 msg="Starting Tempo" version="(version=2.4.0, branch=HEAD, revision=b1c4d3fbf)"
level=warn ts=2024-03-05T14:06:15.134927419Z caller=tempodb.go:623 msg="legacy cache configured with the following roles. Please migrate to the new top level cache configuration." roles="unsupported value type"
level=info ts=2024-03-05T14:06:15.134996091Z caller=cache.go:54 msg="caches available to storage backend" footer=false bloom=true offset_idx=false column_idx=false trace_id_idx=true page=false
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by bpfoster**
### Description of the change

Refactors Tempo memcached configuration to match Tempo v2.4 documentation.

### Benefits

Removes deprecation warning printed to logs and preserves future compatibility when deprecated configurations are removed.

### Possible drawbacks

None noted

### Applicable issues

- fixes #24164

### Additional information

The cache roles are not well documented.  I am assuming these are the safe default values to configure.  When using the old, deprecated configuration, I see this printed to the logs:
`msg="caches available to storage backend" footer=false bloom=true offset_idx=false column_idx=false trace_id_idx=true page=false`
which leads me to believe configuring these roles will preserve the existing behavior.

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/grafana-tempo/Chart.yaml [hunk 1]
```diff
@@ -39,4 +39,4 @@ maintainers:
 name: grafana-tempo
 sources:
 - https://github.com/bitnami/charts/tree/main/bitnami/grafana-tempo
-version: 2.11.0
+version: 2.11.1
```
Comment (FraPazGal) [inline bitnami/grafana-tempo/Chart.yaml:42]: Although debatable, I would say this refactor is not really a fix and going by semver standards we should bump de the minor version of the chart. We also just merged https://github.com/bitnami/charts/pull/24091 so we have to take that into account as well

```suggestion
version: 2.13.0
```

**Review by FraPazGal — COMMENTED** [review]
Hello @bpfoster! I agree with you reasoning and I haven't find anything against using both of the supported roles. I think we con go ahead with that config.

Besides that, I have added a comment as a review. Please check it when you have some time!

**Comment by bpfoster** [comment]
@FraPazGal thank you for the review.  I have updated the chart version per your request!

**Review by FraPazGal — COMMENTED** [review]
It seems our CI tests are not passing @bpfoster, could you please update the Goss test currently failing? The [test looks](https://github.com/bitnami/charts/blob/31069b2010a7324419365e7409a2b896cb32d533/.vib/grafana-tempo/goss/goss.yaml#L19) for an specific line in the `/bitnami/grafana-tempo/conf/tempo.yaml` and your changes are affecting the third regex it looks for:

https://github.com/bitnami/charts/blob/31069b2010a7324419365e7409a2b896cb32d533/.vib/grafana-tempo/goss/goss.yaml#L27

**Comment by bpfoster** [comment]
@FraPazGal thank you for pointing me in the direction of the failure!

Updated the goss test.  I'm not sure what benefit the old matcher provided, looking for a static line.  It now checks that the dynamic memcached cache host is set properly.

**Review by FraPazGal — APPROVED** [review]
Looks perfect now @bpfoster! Thanks again for your contribution!

**Pull request merged**